proposals/0052-fonts.md

```diff
@@ -0,0 +1,369 @@
+**If you have feedback and the feature is released as experimental, please leave it on the Stage 3 PR. Otherwise, comment on the Stage 2 issue (links below).**
+
+- Start Date: 2024-10-15
+- Reference Issues: <!-- related issues, otherwise leave empty -->
+- Implementation PR: <!-- leave empty -->
+- Stage 2 Issue: https://github.com/withastro/roadmap/issues/1037
+- Stage 3 PR: https://github.com/withastro/roadmap/pull/1039
+
+# Summary
+
+Have first-party support for fonts in Astro.
+
+# Example
+
+```js
+// astro config
+export default defineConfig({
+  fonts: {
+    families: ["Roboto", "Lato"],
+  },
+});
+```
+
+```astro
+---
+// layouts/Layout.astro
+import { Font } from 'astro:fonts'
+---
+<head>
+	<Font family='Inter' preload />
+	<Font family='Lato' />
+	<style>
+		h1 {
+			font-family: var(--astro-font-inter);
```

Comment:
Well since it's just some css, I don't think it will cause any issues? In Tailwind 3, you'd have to reference like so:
```js
const defaultTheme = require('tailwindcss/defaultTheme')

module.exports = {
  theme: {
    extend: {
      fontFamily: {
        'sans': ['var(--astro-font-inter)', ...defaultTheme.fontFamily.sans],
      },
    }
  }
}
```

Reply:
Maybe allowthe variable name to be configured in the tag, i.e. `<Font family='Lato' variable='my-custom-font' />` which generates `var(--my-custom-font)`?

This is similar to how [Next.js does it](https://nextjs.org/docs/pages/building-your-application/optimizing/fonts#with-tailwind-css), albeit they use a function syntax instead of a JSX-style component